src/corecel/sys/ScopedProfiling.hh

```diff
@@ -0,0 +1,65 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file corecel/sys/ScopedProfiling.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <string>
+
+#include "celeritas_config.h"
+
+namespace celeritas
+{
+
+//---------------------------------------------------------------------------//
+/*!
+ * Input arguments for the nvtx implementation.
+ */
+struct ScopedProfilingInput
+{
+    // Name of the range
+    std::string name;
+    // ARGB
+    uint32_t color{0xFF00FF00};
+    // User data
+    int32_t payload{0};
+    // Category, used to group ranges together
+    uint32_t category{0};
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * RAII class for scoped profiling.
+ *
+ * Implementations should support multithreaded context where each thread have
+ * one or more alive instance of this class.
+ *
+ * This is useful for wrapping specific code fragment in a range for profiling,
+ * e.g. ignoring of VecGeom instantiation kernels, profiling a specific action
+ * or loop on the CPU.
+ * TODO: Template ScopedProfiling over profiling backend if we need to add a
+ * new one
+ */
+class ScopedProfiling
+{
+  public:
+    using Input = ScopedProfilingInput;
+    // Activate profiling
+    explicit ScopedProfiling(Input input);
```

Comment:
Should we always have to specify all the input parameters, or are they just options? I think some of the code would be cleaner if we also provide a "name-only" constructor for the simple use cases.

Reply:
A constructor with just the name definitely makes sense as most of the time we'd want to only specify that.